.github/workflows/go-ct/action.yml

```diff
@@ -0,0 +1,12 @@
+name: 'Run Component Test for Go SDK'
+runs:
+  using: 'composite'
+  steps:
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Build and run Docker Compose
+      run: |
+        cd ${{ github.workspace }}/ocaml/sdk-gen/component-test
+        docker-compose up --build --exit-code-from test-go-client
```

Comment:
I don't believe that docker composed is needed to run the server and make the client interact with it, some bash script should be enough (https://github.com/xapi-project/xen-api/blob/master/ocaml/message-switch/core_test/basic-rpc-test.sh), same with go compilation (https://github.com/actions/setup-go)

Docker compose is a big hammer, I'd rather have something simpler, that can run as part of tests in the makefile

Reply:
>  It offers several advantages that streamlines the process of building, testing, and simplifying the definition and sharing of multi-container applications for different languages by allowing us to define services in a YAML file and orchestrate them with a single command.

This is what I don't like about this proposal, I don't see a need to define services. This is not live-running system, it's a single run needed for testing. It introduces a new, complex tool, with complex configuration syntax (yaml _is_ complex, unfortunately), but to little benefit, if at all. The shellscript I linked already 'orchestrates' 3 different servers and clients to run the necessary tests.

Reply:
I'm going to give your suggestion a good thought. Prior to making any decision, I need to understand the function of the shell script you linked (make a discussion with NKG toolstack team) and whether it meets our needs for go sdk component testing.

Reply:
Updated the code using shell script instead of docker compose.